main.go

```diff
@@ -77,12 +76,31 @@ type encryptCloudImageBase struct {
 
 	workingDir string
 	devPath    string
+	devPathFormat string
 
 	partitions    gpt.Partitions
 	rootPartition *gpt.PartitionEntry
 	esp           *gpt.PartitionEntry
 }
 
+func (b *encryptCloudImageBase) initDevPathFormat(devPath string) error {
+	b.devPath = devPath
+
+	if strings.HasPrefix(b.devPath, "/dev/nbd") || strings.HasPrefix(b.devPath, "/dev/nvme") {
+		b.devPathFormat = "%sp%d"
+	} else if strings.HasPrefix(b.devPath, "/dev/sd") || strings.HasPrefix(b.devPath, "/dev/vd") {
+		b.devPathFormat = "%s%d"
+	} else {
+		return xerrors.Errorf("Unsuppoted device path: %s. Please look at the code to deterimine what's currently supported.", devPath)
+	}
+
+	return nil
+}
```

Comment:
This function isn't called when the input is a file (where `connectNbd` is called), which breaks `rootDevPath`, `espDevPath` and everything that calls them because `devPathFormat` isn't initialized to anything.

Reply:
Removed format field and calculating it when needed.